### PR TITLE
fix: use nix signal feature to allow for dep upgrade

### LINKED ISSUE #5936: Rust - Investigate and fix breaking changes in upgrading from nix crate 0.26.2 to 0.27.1
The dependabot upgrade is failing because the new version has some breaking changes
https://github.com/build-trust/ockam/actions/runs/6121859822/job/16616475655?pr=5838#step:6:1951

### PR DESCRIPTION
**Pull request by hseeberger**
Ref #5936.

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@hseeberger awesome! Thank you, while we review please

- [ ] Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

**Comment by mrinalwadhwa** [comment]
Nice the change works! There's a minor formatting/spacing error, if you could address that and accept the CLA, we can merge 👍  https://github.com/build-trust/ockam/actions/runs/6189600055/job/16804078484?pr=5974#step:5:610

File: implementations/rust/ockam/ockam_api/Cargo.toml [hunk 1]
```diff
@@ -43,7 +43,7 @@ kafka-protocol = "0.6.1"
 lru = "0.11.1"
 miette = "5.10.0"
 minicbor = { version = "0.19.0", features = ["alloc", "derive"] }
-nix = "0.27"
+nix = {verion = "0.27", features = ["signal"] }
```
Comment (metaclips) [inline implementations/rust/ockam/ockam_api/Cargo.toml]: There's a typo here, `verion`.

**Review by metaclips — CHANGES_REQUESTED** [review]
Little changes and accepting the CLA, and we should merge. Thanks @hseeberger

File: implementations/rust/ockam/ockam_api/Cargo.toml (hunk 1 above)
Comment (hseeberger) [reply implementations/rust/ockam/ockam_api/Cargo.toml]: Bummer!

**Comment by hseeberger** [comment]
Accepted the CLA: https://github.com/build-trust/ockam-contributors/pull/274

File: implementations/rust/ockam/ockam_api/Cargo.toml (hunk 1 above)
Comment (hseeberger) [reply implementations/rust/ockam/ockam_api/Cargo.toml]: Should be fixed now 🤞

Comment (hseeberger) [reply implementations/rust/ockam/ockam_api/Cargo.toml]: I built locally before and it (seemed to) work(ed). Not sure what happened. There are lots of crates, though, and some might not be built when just using `cargo check` form the workspace directory. Anyway, this time I actually built that particular crate, and even tested it: `cargo test -p ockam_api`

**Comment by hseeberger** [comment]
Will close this PR and create a new one against develop!

**Comment by hseeberger** [comment]
See #6006.

**Pull request closed without merging**